Readme.md

```diff
@@ -1,11 +1,26 @@
 # The Big Give's Mautic
 
-This repository is a thin layer for tooling used by the Big Give to
-test and deploy Mautic.
+We use Mautic's official Composer recommendation repository and its Composer
+packages as the basis for building this, but also check in scaffolding
+output from Composer install-ing, to make the behaviour similar to others
+extending `thebiggive/php` and make changes to the scaffolding clear in source
+control history. (Since the `Dockerfile` runs `composer install` and could generate
+these, the process would probably also technically work without checking all
+the scaffolding in.)
```

Comment:
My firefox really struggles to work when viewing the PR that adds all these thousands of files - despite Github's message that " The diff you're trying to view is too large. We only load the first 3000 changed files. ". That makes me think it might be better not to check them in and rely on the composer install. Or if the point so not be reliant on packagist etc for our infrastructure deploys we could check them in as a zip / tar file and then unpack them in the Dockerfile.

How much advantage is there to checking in? Would pinning to a specific version number and then just linking to changes in the upstream be enough to make changes visible in future? Seems like it will often be far more changes than we would want to review anyway so I'm not sure having them visible in our history is useful.

Reply:
@bdsl I don't really know if there's much advantage – mostly just went with this as the path of least resistance having done the install outside Docker. But so long as the `Dockerfile` includes the step which generates scaffolding, as it does already, I don't think there's a critical reason to go with either option.

I don't think we'll ever have to look at the whole diff again, but agree there's not much benefit specifically in tracking our own copy of changes so long as we're not customising anything in `docroot`.

I also share the hunch that it's not the best if we can avoid it anyway, especially as IDEA just froze getting the revert menu item ready for me! So I have reverted the commit adding scaffolding, and brought back an updated version of the readme.

There's a small caveat with the new version which is that re-running the install replaces our `.gitignore`, which in turn makes scaffolding eligible for checking back in. Can't immediately think of a way to stop that happening.

Reply:
@NoelLH can't we just add a command to 'echo >>' a line to the .gitignore or overwrite the .gitignore with our own one copied from outside docroot into the Dockerfile?

Reply:
@bdsl Yep I guess we could copy our own in, in an additional Composer post-install script. I don't think it's a high priority though as it's mostly just a question of remembering not to merge this if we did a local install, which hopefully won't be needed often if/once this works.